odoo/fields.py

```diff
@@ -1331,19 +1332,31 @@ def recompute(self, records):
         if not to_compute_ids:
             return
 
+        def missing_guard(func, batch_records):
+            try:
+                func(batch_records)
+            except MissingError:
+                existing = batch_records.exists()
+                func(existing)
+                for f in batch_records.pool.field_computed[self]:
+                    batch_records.env.remove_to_compute(f, batch_records - existing)
+
         if self.recursive:
-            for record in records:
-                if record.id in to_compute_ids:
-                    self.compute_value(record)
+            def recursive_compute(batch_records):
+                for record in batch_records:
+                    if record.id in to_compute_ids:
+                        self.compute_value(record)
```

Comment:
I would apply `missing_guard` on each record instead.
```suggestion
            for record in batch_records:
                if record.id in to_compute_ids:
                    missing_guard(self.compute_value, record)
```

Reply:
It means that we will make one `exists` by record (instead of batch it) :crying_cat_face: